### PR TITLE
just: add get-credentials-tdxbm

### PR DESCRIPTION
**Pull request by katexochen**

### CODE REVIEW

**Review by msanft — APPROVED** [review]

**Comment by katexochen** [comment]
cc @burgerdev

File: justfile [hunk 1]
```diff
@@ -224,6 +224,18 @@ get-credentials-ci:
         --name "contrast-ci" \
         --admin
 
+get-credentials-tdxbm:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    tmpConfig=$(mktemp)
+    gcloud secrets versions access \
+        projects/796962942582/secrets/m50-ganondorf-kubeconf/versions/2 \
+        --out-file="$tmpConfig"
```
Comment (burgerdev) [inline justfile:233]: Do we really need this in the `justfile`? It sounds like it should be part of Github Actions or something.

File: justfile [hunk 2]
```diff
@@ -224,6 +224,18 @@ get-credentials-ci:
...
+    mergedConfig=$(mktemp)
+    export KUBECONFIG=$tmpConfig:~/.kube/config
+    kubectl config view --flatten > $mergedConfig
+    mv $mergedConfig ~/.kube/config
```
Comment (burgerdev) [inline justfile:237]: This should respect an existing `KUBECONFIG` var.

File: justfile (hunk 1 above)
Comment (katexochen) [reply justfile:233]: It enables devs to quickly setup the needed creds, without setting up ssh keys and searching for kube config in other users home directory. Only requirement is to be in the tailnet (should add a command with this requirement).

Comment (burgerdev) [reply justfile:233]: On the other hand, this is only useful to a particular group of devs - those who have access to this gcp secret. I simply wonder how much org-specific things we should put into OSS, but feel free to go ahead with it.

File: justfile (hunk 2 above)
Comment (katexochen) [reply justfile:237]: I can use 
```sh
export KUBECONFIG=$tmpConfig:${KUBECONFIG:-~/.kube/config}
```
for respecting it as input, but how can I respect it when writing the config? Or is it okay to always update .kube/config, even if KUBECONFIG != .kube/config?

File: justfile [hunk 3]
```diff
@@ -224,6 +224,18 @@ get-credentials-ci:
...
+    #!/usr/bin/env bash
+    set -euo pipefail
+    tmpConfig=$(mktemp)
+    gcloud secrets versions access \
+        projects/796962942582/secrets/m50-ganondorf-kubeconf/versions/2 \
+        --out-file="$tmpConfig"
+    mergedConfig=$(mktemp)
+    export KUBECONFIG=$tmpConfig:${KUBECONFIG:-~/.kube/config}
+    kubectl config view --flatten > $mergedConfig
+    mv $mergedConfig ~/.kube/config
```
Comment (burgerdev) [inline justfile:237]:
```suggestion
    env KUBECONFIG=$tmpConfig:${KUBECONFIG:-~/.kube/config} kubectl config view --flatten > $mergedConfig
    mv $mergedConfig "${KUBECONFIG}"
```

File: justfile (hunk 2 above)
Comment (burgerdev) [reply justfile]: Made a suggestion.

File: justfile (hunk 3 above)
Comment (katexochen) [reply justfile:237]: What if KUBECONFIG=./foo:./bar?

File: justfile [hunk 4]
```diff
@@ -224,6 +224,18 @@ get-credentials-ci:
...
+
```
Comment (katexochen) [inline justfile:238]: Reminder to add a step to set active context.

File: justfile (hunk 3 above)
Comment (burgerdev) [reply justfile:237]: `${KUBECONFIG%%:*}`?

File: justfile [hunk 5]
```diff
@@ -224,6 +224,20 @@ get-credentials-ci:
...
+    tmpConfig=$(mktemp)
+    gcloud secrets versions access \
+        projects/796962942582/secrets/m50-ganondorf-kubeconf/versions/2 \
+        --out-file="$tmpConfig"
+    mergedConfig=$(mktemp)
+    KUBECONFIG_BAK=${KUBECONFIG:-~/.kube/config}
+    KUBECONFIG=$tmpConfig:${KUBECONFIG_BAK} kubectl config view --flatten > $mergedConfig
+    export newContext=$(yq -r '.contexts.[0].name' $tmpConfig)
+    yq -i '.current-context = env(newContext)' $mergedConfig
+    mv $mergedConfig ${KUBECONFIG_BAK%%:*:}
```
Comment (burgerdev) [inline justfile:239]: Did you put the final `:` deliberately? My suggestion was meant to take everything up to exclduing the first `:` char.

**Review by burgerdev — APPROVED** [review]

File: justfile (hunk 5 above)
Comment (katexochen) [reply justfile]: no idea how that slipped in

**Review by burgerdev — APPROVED** [review]

**Pull request merged**